tests/run-macros/quote-inline-function.check

```diff
@@ -3,13 +3,11 @@ Normal function
   var i: scala.Int = 0
   val j: scala.Int = 5
   while (i.<(j)) {
-    val x$1: scala.Int = i
-    f.apply(x$1)
+    f.apply(i)
     i = i.+(1)
   }
   while ({
-    val x$2: scala.Int = i
-    f.apply(x$2)
+    f.apply(i)
```

Comment:
This change looks fishy

Reply:
How so? I figured it would make sense for betaReduce to either behave like `f.appliedToArgs(...)` or a full inlining. (with this case being `appliedToArgs`)

If the other way is preferred, can you explain why so I can know for future PRs?

Reply:
(the other way being "pull all the args out into vals or defs, then either inline or appliedToArgs")